lib/gst/plugin/gstclappersink.c

```diff
@@ -643,6 +704,15 @@ gst_clapper_sink_change_state (GstElement *element, GstStateChange transition)
       gst_element_state_get_name (GST_STATE_TRANSITION_NEXT (transition)));
 
   switch (transition) {
+    case GST_STATE_CHANGE_NULL_TO_READY:
+      /* Reset stream_orientation */
+      self->stream_orientation = GST_VIDEO_ORIENTATION_IDENTITY;
+      if (self->rotation_mode == GST_VIDEO_ORIENTATION_AUTO) {
+        GST_CLAPPER_SINK_LOCK (self);
+        gst_clapper_paintable_set_rotation (self->paintable, self->stream_orientation);
+        GST_CLAPPER_SINK_UNLOCK (self);
+      }
```

Comment:
This whole block should probably be within locked mutex (not only set_rotation).